printing/proof_diffs.ml

```diff
@@ -106,7 +106,7 @@ let tokenize_string s =
     let toks = stream_tok [] lex in
     CLexer.Lexer.State.set st;
     toks
-  with exn ->
+  with exn when CErrors.noncritical exn ->
```

Comment:
This is incorrect, the state should be reset and the exception re-reraised.

Reply:
Thanks. I've added a `Fun.protect` around this. Is the semantics appropriate now?